jbmc/src/java_bytecode/select_pointer_type.cpp

```diff
@@ -78,15 +82,36 @@ pointer_typet select_pointer_typet::specialize_generics(
       return pointer_type;
     }
     const pointer_typet &type =
-      generic_parameter_specialization_map.find(parameter_name)->second.top();
+      generic_parameter_specialization_map.find(parameter_name)->second.back();
+
+    // avoid infinite recursion
+    if(visited_nodes.find(parameter_name) != visited_nodes.end())
+    {
+      optionalt<pointer_typet> result = get_instantiated_type(
```

Comment:
I'm curious, `specialize_generics` is itself recursive, isn't this redundant in some way? Or does this new function solely serves the purpose of breaking cycles? If so, it would be helpful if you could change the name to something that reflects it, e.g., `resolve_specialization_cycle` or something like that.